From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Raine Makelainen <raine.makelainen@jolla.com>
Date: Fri, 13 Nov 2020 10:31:26 +0200
Subject: [PATCH 58/67] Hardcode loopback address for profile lock filename.
 JB#45654

When engine started without network PR_GetHostByName takes 20seconds
when connman tries to resolve host name. As this is only used as part of the
profile lock filename it can as well be like "127.0.0.1:+<pid>".

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 toolkit/profile/nsProfileLock.cpp | 24 ++----------------------
 1 file changed, 2 insertions(+), 22 deletions(-)

diff --git a/toolkit/profile/nsProfileLock.cpp b/toolkit/profile/nsProfileLock.cpp
index 3d17171117c8..f2c006d38e30 100644
--- a/toolkit/profile/nsProfileLock.cpp
+++ b/toolkit/profile/nsProfileLock.cpp
@@ -309,29 +309,9 @@ nsresult nsProfileLock::LockWithSymlink(nsIFile* aLockFile,
   struct in_addr inaddr;
   inaddr.s_addr = htonl(INADDR_LOOPBACK);
 
-  // We still have not loaded the profile, so we may not have proxy information.
-  // Avoiding a DNS lookup in this stage makes sure any proxy is not bypassed.
-  // By default, the lookup is enabled, but when it is not, we use 127.0.0.1
-  // for the IP address portion of the lock signature.
-  // However, this may cause the browser to refuse to start in the rare case
-  // that all of the following conditions are met:
-  //   1. The browser profile is on a network file system.
-  //   2. The file system does not support fcntl() locking.
-  //   3. The browser is run from two different computers at the same time.
-#  ifndef MOZ_PROXY_BYPASS_PROTECTION
-  char hostname[256];
-  PRStatus status = PR_GetSystemInfo(PR_SI_HOSTNAME, hostname, sizeof hostname);
-  if (status == PR_SUCCESS) {
-    char netdbbuf[PR_NETDB_BUF_SIZE];
-    PRHostEnt hostent;
-    status = PR_GetHostByName(hostname, netdbbuf, sizeof netdbbuf, &hostent);
-    if (status == PR_SUCCESS) memcpy(&inaddr, hostent.h_addr, sizeof inaddr);
-  }
-#  endif
-
   mozilla::SmprintfPointer signature =
-      mozilla::Smprintf("%s:%s%lu", inet_ntoa(inaddr),
-                        aHaveFcntlLock ? "+" : "", (unsigned long)getpid());
+      mozilla::Smprintf("127.0.0.1:%s%lu", aHaveFcntlLock ? "+" : "",
+                        (unsigned long)getpid());
   const char* fileName = lockFilePath.get();
   int symlink_rv, symlink_errno = 0, tries = 0;
 
